typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,36 @@
+const $tabContainer = document.querySelector('.tab-container');
+const $allTabs = document.querySelectorAll('.tab');
+const $allViews = document.querySelectorAll('.view');
+
+if (!$tabContainer) throw new Error('$tabContainer query failed');
+if (!$allTabs) throw new Error('$allTabs query failed');
+if (!$allViews) throw new Error('$allViews query failed');
+
+$tabContainer.addEventListener('click', (event: Event) => {
+  const $eventTarget = event.target as HTMLElement;
+  if (!$eventTarget) throw new Error('$focusTab query failed');
+
+  if (!$eventTarget.matches('.tab')) {
+    return;
+  }
+
+  for (let i = 0; i < $allTabs.length; i++) {
+    if ($eventTarget === $allTabs[i]) {
+      $allTabs[i].className = 'tab active';
+    } else {
+      $allTabs[i].className = 'tab';
+    }
+  }
+
+  for (let i = 0; i < $allViews.length; i++) {
+    $allViews[i].className = 'view hidden';
+  }
+
+  const currentView = $eventTarget.dataset.view;
+
+  for (let i = 0; i < $allViews.length; i++) {
+    if ($allViews[i].getAttribute('data-view') === currentView) {
+      $allViews[i].className = 'view';
+    }
+  }
```

Comment:
You should try to combine these two `for` loops into one `for` loop. You are looking over the same item, in the same direction. Try to make it more efficient!

Reply:
I was trying to ensure that one view wouldn't appear before the current view was hidden. Is that not really something to be concerned about here?